src/main/java/bio/terra/service/tabulardata/azure/StorageTableDao.java

```diff
@@ -0,0 +1,131 @@
+    var tableClient = tableServiceClient.getTableClient(azurfyTableName(datasetName));
+    ListEntitiesOptions options =
+        new ListEntitiesOptions().setFilter(String.format("PartitionKey eq '%s'", loadTag));
+    var pagedEntities = tableClient.listEntities(options, null, null);
+    // This could be more efficient, but would need to implement some sort of index to query on.
+    return pagedEntities.stream()
+        .skip(offset)
+        .limit(limit)
+        .map(StorageTableDao::storageTableEntityBulkFileLoadModel)
+        .collect(Collectors.toList());
+  }
+
+  private static TableEntity bulkFileLoadModelToStorageTableEntity(
+      BulkLoadHistoryModel model, String loadTag, Instant loadTime) {
+    return new TableEntity(loadTag, model.getFileId())
+        .addProperty("load_tag", loadTag)
+        .addProperty("load_time", loadTime)
+        .addProperty("source_name", model.getSourcePath())
+        .addProperty("target_path", model.getTargetPath())
+        .addProperty("state", model.getState().name())
+        .addProperty("file_id", model.getFileId())
+        .addProperty("checksum_crc32c", model.getChecksumCRC())
+        .addProperty("checksum_md5", model.getChecksumMD5())
+        .addProperty("error", model.getError());
+  }
+
+  private static BulkLoadHistoryModel storageTableEntityBulkFileLoadModel(TableEntity tableEntity) {
+    var model =
+        new BulkLoadHistoryModel()
+            .sourcePath(tableEntity.getProperty("source_name").toString())
+            .targetPath(tableEntity.getProperty("target_path").toString())
+            .state(BulkLoadFileState.valueOf(tableEntity.getProperty("state").toString()))
+            .fileId(tableEntity.getProperty("file_id").toString());
+
+    Optional.ofNullable(tableEntity.getProperty("checksum_crc32c"))
+        .ifPresent(o -> model.checksumCRC(o.toString()));
+    Optional.ofNullable(tableEntity.getProperty("checksum_md5"))
+        .ifPresent(o -> model.checksumMD5(o.toString()));
+    Optional.ofNullable(tableEntity.getProperty("error")).ifPresent(o -> model.error(o.toString()));
+    return model;
+  }
+
+  /**
+   * Generate a valid azure storage table name for load history. The resulting name will be the
+   * input truncated to fit {@value #LOAD_HISTORY_TABLE_NAME_SUFFIX} as a suffix in less than 63
+   * characters, with all non-alphanumeric characters stripped out.
+   *
+   * @param tableName The root of the table name
+   * @return A valid azure storage table name with load history suffix.
+   */
+  private static String azurfyTableName(String tableName) {
+    var alphaNumeric = tableName.replaceAll("[^A-Za-z0-9]", "");
+    final String rightLength;
+    if (alphaNumeric.length() <= TABLE_NAME_MAX_LENGTH_BEFORE_PREFIX) {
+      rightLength = alphaNumeric + LOAD_HISTORY_TABLE_NAME_SUFFIX;
+    } else {
+      rightLength =
+          alphaNumeric.substring(0, TABLE_NAME_MAX_LENGTH_BEFORE_PREFIX)
+              + LOAD_HISTORY_TABLE_NAME_SUFFIX;
+    }
```

Comment:
I think this is 
```suggestion
    String  rightLength =  alphaNumeric.substring(0, Math.min(alphaNumeric.length(),
        TABLE_NAME_MAX_LENGTH_BEFORE_PREFIX)) + LOAD_HISTORY_TABLE_NAME_SUFFIX;
```
If you don't care which end you truncate from, you could use `String.substring(int)` instead of `substring(int, int)`